consul/acl_replication.go

```diff
@@ -0,0 +1,351 @@
+package consul
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/hashicorp/consul/consul/structs"
+	"github.com/hashicorp/consul/lib"
+)
+
+// aclIDSorter is used to make sure a given list of ACLs is sorted by token ID.
+// This should always be true, but since this is crucial for correctness and we
+// are accepting input from another server, we sort to make sure.
+type aclIDSorter struct {
```

Comment:
You could probably alias `structs.ACLs` directly instead of having a parent struct